src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubHubResource.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Provisioning.WebPubSub;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure Web PubSub Hub setting resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="webpubsub">The <see cref="AzureWebPubSubResource"/> that the resource belongs to.</param>
+public class AzureWebPubSubHubResource(string name, AzureWebPubSubResource webpubsub) : AzureConstructResource(name, s_empty),
```

Comment:
This is a resource but it shouldn't be an `AzureConstructResource`.

Reply:
If it is not an AzureConstructResource, it is not allowed to use "ConfigureConstruct" to customize the hub properties as shown in the test case : 
```cs

        wps.AddHub("abc").ConfigureConstruct(construct =>
        {
            var hub = construct.GetResources().OfType<WebPubSubHub>().First(i => i.Name.Value == "abc");
            hub.Properties.Value!.AnonymousConnectPolicy = "allow";
        });
        ```

Reply:
I see. Why make it empty by default the?

Reply:
because the WebPubSubHub has a parent service "WebPubSubService", which is only known in the WebPubSubResource's construct.....

Reply:
Ah I see, let me change it to get the parent service from construct

Reply:
And `GetBicepTemplateFile` is not expected to be supported by `AzureWebPubSubHubResource` as it is a subresource

Reply:
This is a very nice pattern. I'll think about where we can apply this elsewhere (probably event hubs as well).

Reply:
Spoke to @eerhardt, the user can customize the construct using the AddAzureWebPubSub and this resource doesn't need to be an AzureConstructResource.

```C#
wps.ConfigureConstruct(construct =>
{
    var hub = construct.GetResources().OfType<WebPubSubHub>().First(i => i.Name.Value == "abc");
    hub.Properties.Value!.AnonymousConnectPolicy = "allow";
});
```

Reply:
> construct.GetResources().OfType<WebPubSubHub>().First(i => i.Name.Value == "abc");

FYI - in general this needs to be `IdentifierName` as sometimes `Name` gets autogenerated with a unique string.

Reply:
But IdentifierName is get normalized with `Infrastructure.NormalizeIdentifierName` though, it is not necessarily the name user passed in?